### PR TITLE
Update Reference Documentation

### PR DESCRIPTION
**Pull request by b-studios**
This PR updates the reference documentation:

- It mostly replaces "Dotty" -> "Scala 3"
- It updates some old `using` Syntax

### CODE REVIEW

**Comment by b-studios** [comment]
@nicolasstucki I tagged you as a reviewer, since besides the trivial changes from Dotty to Scala 3, I also make small changes to the meta programming docs.

File: docs/docs/reference/metaprogramming/tasty-reflect.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ def natConstImpl(x: Expr[Int])(using Quotes): Expr[Int] = {
   import quotes.reflect._
   val xTree: Term = x.asTerm
   xTree match {
-    case Inlined(_, _, Literal(Constant(n: Int))) =>
+    case Inlined(_, _, Literal(Constant.Int(n))) =>
```
Comment (nicolasstucki) [inline docs/docs/reference/metaprogramming/tasty-reflect.md:47]:
```suggestion
    case Inlined(_, _, Literal(IntConstant(n))) =>
```

**Review by nicolasstucki — APPROVED** [review]
Just a small update to the reflect API is needed.

File: docs/docs/reference/metaprogramming/tasty-reflect.md (hunk 1 above)
Comment (b-studios) [reply docs/docs/reference/metaprogramming/tasty-reflect.md]: Thanks! I tested it with M2, but wasn't sure about the current state of affairs.

Comment (nicolasstucki) [reply docs/docs/reference/metaprogramming/tasty-reflect.md]: That one changed in M3 3 days ago. Easy to miss.

**Pull request merged**